### DISCUSSION TITLE
How to tell when a panel is hovered?

### DISCUSSION THREAD

**Discussion by coolcatcoder** [emilk/egui#4996] [Q&A] [answered]
I have a SidePanel, and I want to know when the cursor is hovering over it, or clicked on it, so that I can stop the mouse input from being used else where.
I've tried memory.focused().is_some(), but because a SidePanel isn't a widget, it doesn't work.
I've tried ui.ui_contains_pointer(), but this only works on half the panel, for some reason. (And not the half that contains my widgets...)
I've tried ui.interact_bg(Sense::click()).hovered(), but same issue as the previous attempt.
I've tried ui.interact(ui.max_rect(), ui.id(), Sense::click()).hovered(), but it doesn't detect me hovering above widgets, only above the panel.
The last attempt was the best, but I don't want to || it against every widgets' hover sense, as that would be annoying. Is there a better solution?

**Comment by YgorSouza** [comment]
I'm not sure why `ui.ui_contains_pointer()` isn't working in your case, as it seems to work fine in the quick test I did, but you can try `ui.rect_contains_pointer(ui.max_rect())` (i.e., test with `max_rect()` instead of `min_rect()`). They both do the same for me though.

```rs
impl eframe::App for MyApp {
    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
        egui::SidePanel::left("left").show(ctx, |ui| {
            ui.painter()
                .rect_filled(ui.min_rect(), 0.0, Color32::RED.gamma_multiply(0.3));
            ui.painter()
                .rect_filled(ui.max_rect(), 0.0, Color32::BLUE.gamma_multiply(0.3));
            let min_contains = ui.ui_contains_pointer();
            let max_contains = ui.rect_contains_pointer(ui.max_rect());
            ui.label(format!("{:?}", min_contains));
            ui.button(format!("{:?}", max_contains));
        });
        egui::CentralPanel::default().show(ctx, |ui| {
            ui.label("Central");
        });
    }
}
```

https://github.com/user-attachments/assets/22a9234e-3fd1-4924-9978-ea67aef82f95

**Reply by coolcatcoder** [reply]
That worked perfectly! Thank you so much!

**Reply by coolcatcoder** [reply]
I thought it worked perfectly.... It turns out that if you hover over a ComboBox in the panel, then it thinks you aren't hovering over the panel.

**Reply by coolcatcoder** [reply, marked as answer]
Aha by ||ing your strategy with memory.any_popup_open() it now works hopefully perfectly!
